internal/provider/provider.go

```diff
@@ -128,6 +129,9 @@ func (p *segmentProvider) Configure(ctx context.Context, req provider.ConfigureR
 			URL: url,
 		},
 	}
+	retryClient := retryablehttp.NewClient()
+	retryClient.RetryMax = 10
+	configuration.HTTPClient = retryClient.StandardClient()
```

Comment:
custom client is enabled by [this in the public-api go sdk](https://github.com/segmentio/public-api/blob/master/scripts/generate-sdk/go/templates/client.mustache#L64-L67)